crates/macro-impl/src/html_tree/html_tag/mod.rs

```diff
@@ -185,6 +187,35 @@ impl HtmlTag {
     }
 }
 
+struct HtmlSelfClosingTag;
+
+impl Peek<Ident> for HtmlSelfClosingTag {
+    fn peek(cursor: Cursor) -> Option<Ident> {
+        let (punct, cursor) = cursor.punct()?;
+        (punct.as_char() == '<').as_option()?;
+
+        let (ident, mut cursor) = cursor.ident()?;
+        (ident.to_string().to_lowercase() == ident.to_string()).as_option()?;
+
+        loop {
+            if let Some((punct, next_cursor)) = cursor.punct() {
+                match punct.as_char() {
+                    '/' => return Some(ident),
```

Comment:
I don't think this will work if you have a tag like:

```
<div data-val={2 / 1}>
  ...
</div>
```

Reply:
Right, I forgot about `{ expr }` :sweat_smile: I think that looking for `/>` instead works since this pattern should not appear anywhere else.

Reply:
I didn't notice that braces are handled as groups in `syn`, rather than punctuation, so that should be fixed.

Reply:
Div inside `{ expr }` is fixed, as well as:

```html
<div>
    <div onblur=|_| 2 / 1/>
</div>
```

But this erroneous one provokes a compilation failure by going out of memory:

```html
<div>
    <div onblur=|_| 2 / 1>
</div>
```

Reply:
Fixed.